web/Services/PostcodeService.cs

```diff
@@ -41,15 +64,16 @@ public async Task AddPostcodesToDatabaseIfAbsent(IEnumerable<string> postcodes)
 
         var addPostcodeChunkToResponseList = postcodesToLookupChunks.Select(async p => postcodeResponseList.AddRange(await GetPostcodeDetails(p)));
         await Task.WhenAll(addPostcodeChunkToResponseList);
-        
+
         var postcodeGroups = postcodeResponseList
-                .GroupBy(p => p.Result?.Lat == null || p.Result.Lon == null || p.Result.Postcode == null)
-                .ToDictionary(grouping => grouping.Key ? "invalid" : "valid", grouping => grouping.ToArray());
-    
+            .GroupBy(p => p.Result?.Lat == null || p.Result.Lon == null || p.Result.Postcode == null)

```

Comment:
Github seems to think these `null` checks will always be false as these are `double` not `double?`